src/pymor/vectorarrays/interfaces.py

```diff
@@ -640,11 +640,14 @@ class VectorSpaceInterface(ImmutableInterface):
     is_scalar
         Equivalent to
         `isinstance(space, NumpyVectorSpace) and space.dim == 1 and space.id is None`.
+    dtype
+        type hint to create appropriate `zeros`, `ones`, etc.
```

Comment:
Not sure what you mean by `type hint`. Are you saying that a fixed `VectorSpace` could contain arrays of different `dtype` and this attribute only relates to these methods?

Reply:
If this is its only purpose I see no reason to have it as an attribute of `VectorSpace`. An optional argument to these methods would be sufficient.

Also note that such an interpretation of this `dtype` argument won't help in #669, which caused @pmli to open #670.

Reply:
Well, not _currently_ enforced, as in what I've implemented does not enforce/use the dtype outside of the creation functions. And now `dot`. 
Adding it to all creation functions I find unattractive. Then you'd always have to manually detect which dtype to use before.